mining_test.go

```diff
@@ -152,6 +152,122 @@ func TestTxFeePrioHeap(t *testing.T) {
 	}
 }
 
+// TestTxStakeSortingWithSizeHeap ensures the priority queue for stake
+// transactions, sorting by stake priority, ticket relative absolute
+// fee for small tickets, relative fee per ticket size, then lower
+// stake priority by relative fee per transaction size works as expected.
+func TestTxStakeSortingWithSizeHeap(t *testing.T) {
+	// Create some fake priority items that exercise the expected sort
+	// edge conditions.
+	testItems := []*txPrioItem{
+		{feePerKB: 5678, priority: 5, txType: stake.TxTypeSStx, txSize: 100},
+		{feePerKB: 2840, priority: 2, txType: stake.TxTypeSStx, txSize: 200}, // Higher absolute fees despite size
+		{feePerKB: 1234, priority: 3, txType: stake.TxTypeSStx, txSize: 3000},
+		{feePerKB: 1235, priority: 1, txType: stake.TxTypeSStx, txSize: 3000}, // Too big size, higher fee per KB
+		{feePerKB: 5678, priority: 1, txType: stake.TxTypeSSGen, txSize: 100}, // Votes go first
+		{feePerKB: 5678, priority: 1, txType: stake.TxTypeSSGen, txSize: 200},
+		{feePerKB: 1234, priority: 2, txType: stake.TxTypeSStx, txSize: 100},
+		{feePerKB: 1234, priority: 5, txType: stake.TxTypeSStx, txSize: 100},     // Higher priority ticket all else accounted for
```

Comment:
`txPQByStakeSizeAndFeeAndThenPriority` will not get to looking at `.priority` in the comparison of the above 2 items since it goes into the `if !bothAreLowStakePriority {` block (`bothAreLowStakePriority` is `false` since these are SSTx) and returns `pq.items[i].fee > pq.items[j].fee`.  Is this meant to test priority 5 vs. 2?